### DISCUSSION TITLE
habitat models - include waterbody centrelines as potential rearing habitat for additional species (CH, BT, RB, WCT)

### DISCUSSION THREAD

**Discussion by NewGraphEnvironment** [smnorris/bcfishpass#243] [Ideas]
Because we have commonly been using linear length to filter and report on opportunities, I feel this should be a standard feature vs a branch tweak.  Including it allows a simpler (and less error prone) interpretation of intrinsic value.  All the listed species rear in these areas so if there is modelled habitat connected the centreline length should be included to make comparisons between rearing systems simple.

Having the freshwater atlas waterbody and edge types within the [parameters](https://github.com/smnorris/bcfishpass/tree/main/parameters) files vs hardcoded in numerous scripts is a related idea (not sure how practical though).

An example of edge codes potentially suitable for this are [here](https://github.com/NewGraphEnvironment/bcfishpass/blob/pars2022ng/scripts/channel_width/sql/channel_width.sql)   and look like this

```
 (wb.waterbody_type = 'R'
    OR (wb.waterbody_type IN ('L','W') AND s.edge_type IN (1000,1050,1100,1200,1475,2000,2300))
    OR (wb.waterbody_type IS NULL AND s.edge_type IN (1000,1050,1100,1200,1475,2000,2300)))
```

**Comment by smnorris** [comment]
As background, waterbodies are currently only included as rearing habitat for two species:
- `CO`  - wetlands with connectivity to modelled spawning
- `SK`  - lakes/reservoirs above area threshold noted in parameters

**Comment by BRebellato** [comment]
Chinook don't often rear in lakes. BT, RB and WCT do, but they would need to be adfluvial, which isn't always the case. Suggest adding this measurement in as a separate field from the existing one that looks only at the stream lengths, so that it can be included in habitat estimates when suitable and excluded when not relevant.

**Comment by smnorris** [comment]
Adding a rearing model column `rearing_model_<spp_b>` is straightforward, but there are also the derived columns:
- `<spp_b>_rearing_belowupstrbarriers_km`
- `all_rearing_km`
- `all_rearing_belowupstrbarriers_km`
- `all_spawningrearing_km`
- `all_spawningrearing_belowupstrbarriers_km`

There would have to be two sets of the `all_` columns, or a method to switch what they include.

I'll look into adding this as a parameter. Shouldn't be too hard - the trickier part may be the data management for multiple outputs/scenarios.
